snpm_cp.m

```diff
@@ -782,6 +798,45 @@ function snpm_cp(CWD)
 %-Loop over planes (breaks out after first loop if bVolm)
 %-----------------------------------------------------------------------
 nP = [];
+
+if(UseRapidPT >= 1)
+    
+    params.N = size(X,1);
+    params.V = size(X,2);
+    params.nPerm = nPerm;
+    params.xdim = xdim; 
+    params.ydim = ydim; 
+    params.zdim = zdim; 
+    params.origin = ORIGIN; 
+    params.nGroup1 = size(find(iCond == 1),2); 
+    params.nGroup2 = params.N - params.nGroup1; 
+
+    datasetInfo = strcat(num2str(params.N),'_',num2str(params.nGroup1),'_',num2str(params.nGroup2));
+    runInfo = strcat(datasetInfo,'_',num2str(nPerm),'.mat');
+    save(strcat('params',runInfo),'params');
+
+    
+    fullpath = mfilename('fullpath');
+    % We only need the directory of snpm_cp so we remove the 7 chars
+    % associated to 'snpm_cp' in fullpath.
+    RapidPT_path = strcat(fullpath(1:end-7),'RapidPT_min');
+    addpath(RapidPT_path);
+
+    write = 0;
+    [outputs, timings] = TwoSampleRapidPT(X, nPerm, params.nGroup1, write, RapidPT_path);
+    MaxT = outputs.MaxT;
+    save(strcat('timings',runInfo),'timings');
+
+    % Save variables for snpm_pp
+    MaxT = [MaxT',-MaxT'];
+    % Calculate uncorrected p-vals
+    [~, SnPMucp, ~, ~] = ttest2(X(1:params.nGroup1, :), X(params.nGroup1+1:end, :), 0.05, 'both', 'unequal');
+    save('SnPMucp.mat','SnPMucp')
```

Comment:
Hi @nicholst,

I am trying to figure out where to end the if/else statement that decides to run RapidPT or SnPM. I was thinking in adding it around line 1109 in snpm_cp:

https://github.com/felipegb94/SnPM-devel/blob/master/snpm_cp.m#L1099-L1111

So, right after the variable `nP` is divided by the number of permutations. Since I don't keep track of `nP` in my code I need to figure out how to compute it. When `nP` is defined:

```
nP    = zeros(1,WorkDim);	%-Nonparam P's
```

Does that mean that those are the nonparametric p-values? Are those the corrected p-values? I am not sure if they are because `SnPMucp` is set to `nP` and then saved in line 1109.

Thanks!